homeassistant/components/tuya/climate.py

```diff
@@ -187,9 +232,63 @@ def supported_features(self):
     @property
     def min_temp(self):
         """Return the minimum temperature."""
-        return self._tuya.min_temp()
+        min_temp = self._tuya.min_temp()
+        if min_temp is not None:
+            if self._tuya.has_decimal():
+                return min_temp
+            if min_temp != 0:
+                return min_temp
+        return super().min_temp
 
     @property
     def max_temp(self):
         """Return the maximum temperature."""
-        return self._tuya.max_temp()
+        max_temp = self._tuya.max_temp()
+        if max_temp is not None:
+            if self._tuya.has_decimal():
+                return max_temp
+            if max_temp != 100:
+                return max_temp
+        return super().max_temp
+
+    def _get_ext_temperature(self):
+        if not self._temp_entity:
+            return None
+
+        def _log_error(error_msg):
+            if not self._temp_entity_error:
+                _LOGGER.warning(
+                    "Error on Tuya external temperature sensor %s: %s",
+                    self._temp_entity,
+                    error_msg,
+                )
+                self._temp_entity_error = True
+
+        try:
+            attr_list = self._temp_entity.split(".", 2)
+            if len(attr_list) < 2:
+                _log_error("entity name is invalid")
+                return None
+            entity_name = f"{attr_list[0]}.{attr_list[1]}"
+            entity_attr = None
+            if len(attr_list) > 2:
+                entity_attr = attr_list[2]
+
+            state_obj = self.hass.states.get(entity_name)
```

Comment:
This looks like its re-implementing part of `generic_theromostat` https://www.home-assistant.io/integrations/generic_thermostat/

Is there no way to get the temperature from the tuya api for the climate device?

Reply:
Temperature come from device by default, but some device do not provide this info. So I add the possibility to configure an external sensor to provide the missing information. This is one of the additional configuration option in the new schema.

Reply:
Is the external sensor provided by the integration or another integration?

Reply:
Any sensor in HA.